app/models/activity.py

```diff
@@ -42,22 +43,10 @@
 }
 
 
+@dataclass(init=True, repr=True, unsafe_hash=True)
```

Comment:
Why unsafe_hash?

AFAIK, init and repr are True by default as well

Reply:
Whenever these SQLalchemy objects are created and called, an error shows up indicating the table object to be unhashable. Unless and until it is enabled,  a __hash__() method is generated according to how eq and frozen are set.(which are not set here).

Reply:
And yeah, `init` & `repr` are True by default.